src/apd.go

```diff
@@ -0,0 +1,108 @@
+package src
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"time"
+)
+
+const (
+	defaultBroadCastIP = "255.255.255.255"
+)
+
+type Packet struct {
+	PublicKey string
+	IP        string
+}
+
+type Apd struct {
+	PublicKey string
+	LocalIP   string
+	Port      int
+	PacketMap map[string]string
+	DoneCh    chan error
+	PacketCh  chan Packet
+}
+
+// NewApd returns an Apd type
+func NewApd() *Apd {
+	return &Apd{
+		Port:      3000,
```

Comment:
Since the port and the length of PacketCh are constant for all instances of `APD,` extracting a constant for `port` and `PacketCh` length will make  `APD` more defined to its different instances.